### PR TITLE
testutil: fix WithExperimental also setting "init"

### PR DESCRIPTION
**Pull request by thaJeztah**
Looks like this was overlooked in the review of the PR that added this; e401b88e59e098745744917c555d549f08353e6d (https://github.com/moby/moby/pull/37183)

There is a separate option for `WithInit`, so this option should not automatically enable it when starting  a daemon with experimental enabled.

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @vdemeester @cpuguy83 @SamWhited ptal

**Review by SamWhited — APPROVED** [review]
LGTM assuming tests that used this pass.

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Comment by thaJeztah** [comment]
Seeing a failure on ppc64le https://ci.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-40012/runs/1/nodes/241/log/?start=0

Don't think it's related, but the behaviour with `--init` (w.r.t. exit codes) could be slightly different, so triggering again;


```
 === FAIL: ppc64le.integration.service TestCreateServiceSecretFileMode (7.72s)
     create_test.go:224: Creating a new daemon at: "/go/src/github.com/docker/docker/bundles/test-integration/TestCreateServiceSecretFileMode"
     create_test.go:265: assertion failed: 2 (int) != 1 (int)
     create_test.go:275: assertion failed: string "" does not contain "-rwxrwxrwx"
```


Failure on Windows RS5 looks a flaky (but is something we should have a look at);
https://ci.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-40012/runs/1/nodes/182/log/?start=0


```
 --- FAIL: TestDockerSuite/TestRunInteractiveWithRestartPolicy (4.29s)
     docker_cli_run_test.go:1797: assertion failed: 
         Command:  d:\CI\PR-40012\1\binary\docker.exe run -i --name test-inter-restart --restart=always busybox sh
         ExitCode: 0
         Stdout:   
         Stderr:   
         
         Failures:
         ExitCode was 0 expected 11
```

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**
